fix: maker command (generator stubs)

raffaelecarelle opened the pull request:
### 1. Why is this change necessary?
This changes going to fix maker command (stubs generator)

### 2. What does this change do, exactly?
Removed scafolding collector that collect all generator stubs (not only one)
and generated stubs for only maker choosed

### 3. Describe each step to reproduce the issue or behaviour.
Create a plugin
install and activated plugin (skip all stub generator)
change a little bit composer.json file
do bin/console make:plugin:controller
after generated controller, composer.json file was restored, because not only maker of controller stub is generated but all of them.

### 4. Please link to the relevant issues (if any).

raffaelecarelle committed:
Fix maker command appling generate stubs for only command choosed

AydinHassan commented:
hey @raffaelecarelle what happens now when you use a generator which adds config? Does it now overwrite the config file with broken xml? Eg the `services.xml` would just have the entry rather than the header/footer?

raffaelecarelle commented:
> hey @raffaelecarelle what happens now when you use a generator which adds config? Does it now overwrite the config file with broken xml? Eg the `services.xml` would just have the entry rather than the header/footer?

Hi @AydinHassan, good point, you're right. services.xml will be override with only generator config stubs. For now, it would let user modify services.xml manually I think. it seems better of nothing

AydinHassan commented:
I think I would want to see a solution where we patch the files if they exist, or at a minimum, if the files exist, just display the inner content to the user on the cli so they can patch it themselves?

raffaelecarelle commented:
> I think I would want to see a solution where we patch the files if they exist, or at a minimum, if the files exist, just display the inner content to the user on the cli so they can patch it themselves?

for patching we need a xml manipulator that can add "service" node. It's fine for me.
second solution is faster but less cool...

raffaelecarelle committed:
Add assertions for configuration hints in MakerCommandTest

Updated MakerCommandTest to include assertions that check for specific configuration hints in the command output. Additionally, appended 'src/Resources/config/services.xml' with dummy content to improve testing completeness.

raffaelecarelle committed:
fix cs

raffaelecarelle commented:
Hi @AydinHassan, I thought the best solution is to print on the screen the configuration to add manually to services/route and so on. Because users can use other types of configurations like php or yaml. And then there is the case of no one configuration file with Autowire and Autoconfiguration.

raffaelecarelle committed:
fix cs

raffaelecarelle committed:
Update TestsGenerator to use Filesystem mock

This commit updates the `TestsGenerator` class and its corresponding test suite to use a mocked `Filesystem`. By ensuring the `Filesystem` dependency is injected and mocked in tests, we improve the testability and reliability of the code.

raffaelecarelle committed:
scaffolding provider has always option now

AydinHassan commented on src/Core/Framework/Plugin/Command/MakerCommand.php:
```diff
@@ -100,19 +100,48 @@ protected function execute(InputInterface $input, OutputInterface $output): int
                 $directory
             );
 
+            if (\method_exists($this->generator, 'disableAskConfirmQuestion')) {
```
What is this change doing? What are you trying to fix here? We typically don't like to manipulate state like this so it would be good to understand what is happening here.

raffaelecarelle replied:
Yes. This I add disableAskConfirmQuestion method due to disable $io->confirm(self::CLI_QUESTION). On creation plugin is usefull but not in MakerCommand because if a user wants to create something specific, the question is no longer needed. Have you any suggest to do better? I should add  disableAskConfirmQuestion method on GeneratorInterface?

AydinHassan replied:
maybe we can add a prop to `PluginScaffoldConfiguration` ? like `shouldAskQuestions`. Then in `\Shopware\Core\Framework\Plugin\Command\MakerCommand` we set it to false. Then the generators can just query the config?  cc @jozsefdamokos maybe you have some opinion :)

AydinHassan commented on src/Core/Framework/Plugin/Command/MakerCommand.php:
```diff
@@ -100,19 +100,48 @@ protected function execute(InputInterface $input, OutputInterface $output): int
+        return $configStubs;
+    }
+
+    private function isXml(string $filename): bool
```
This looks unused now.

AydinHassan commented on src/Core/Framework/Plugin/Command/MakerCommand.php:
```diff
@@ -100,19 +100,48 @@ protected function execute(InputInterface $input, OutputInterface $output): int
 
             $io->success('Scaffold created successfully');
 
+            foreach ($configsStubs as $stub) {
```
So now we never write the config files even if they don't exist? Honestly I don't think this is a good dev experience. Imo, if it does not exists we should write it, if it exists show the new content that needs to be manually applied. And I guess it's the same for any file, we don't need to distinguish between config files with this path matching you have?

raffaelecarelle committed:
Remove unused isXml function

The isXml function was removed from the MakerCommand class as it was no longer being used. This cleanup improves code maintainability by eliminating redundant methods.

mitelg commented:
hey @raffaelecarelle 

as @AydinHassan is currently out of office, I am wondering what the status is here? :slightly_smiling_face:

mitelg commented:
Hey @raffaelecarelle 

This PR was closed due to inactivity. If this change is still important to you, feel free to create a new pull request. 

For more information about our contribution guidelines, see https://github.com/shopware/shopware/blob/trunk/CONTRIBUTING.md
